src/components/kytos/napp/Toolbar.vue

```diff
@@ -11,110 +11,43 @@
- import * as Vue from 'vue'
- 
- const options = {
- 
-   moduleCache: {
-     vue: Vue
-   },
-
-   pathResolve({ refPath, relPath }, options) {
-
-    if ( relPath === '.' )
-      return refPath;
-
-    if ( relPath[0] !== '.' && relPath[0] !== '/' )
-      return relPath;
-
-    return String(new URL(relPath, refPath === undefined ? window.location : refPath));
-   },
-
-   handleModule: async function (type, getContentData, path, options) {
-
-    switch (type) {
-      case '.css':
-        options.addStyle(await getContentData(false));
-        return null;
-      case '.kytos':
-        console.log("Kytos extension detected. Switch extension to .vue");
-        return null;
-      default: return undefined; // let vue3-sfc-loader handle this
-    }
-   },
-
-   async getFile(url) {
- 
-     const res = await fetch(url);
-     if ( !res.ok )
-        throw Object.assign(new Error(url+' '+res.statusText), { res });
-     return await {type: '.vue', getContentData: ()=>res.text()};
-   },
- 
-   addStyle(textContent) {
- 
-     const style = Object.assign(document.createElement('style'), { textContent });
-     const ref = document.head.getElementsByTagName('style')[0] || null;
-     document.head.insertBefore(style, ref);
-   },
-
-   log(type, ...args) {
-     
-     console.log("vue3-sfc-loader log:");
-     console[type](...args);
-   }
- 
- }
+ import axios from 'axios'
+ import { mapState } from 'pinia'
+ import { useToolbarStore } from '../../../stores/toolbarStore'
  
  export default {
    name: 'k-toolbar',
    props: ["active", "compacted", "components"],
    data () {
      return {
-      url: this.$kytos_server+ 'ui/k-toolbar',
-      template: null,
-      inner_components: this.components || [] ,
-      data: [],
-      parsed: ""
+      url: this.$kytos_server + 'ui/k-toolbar',
      }
    },
-   render: function(createElement){
-     if (this.template) return this.template();
-   },
    created() {
-       var self = this
-       $.get({
-         url: this.url,
-         datatype: 'json',
-         async: true,
-         cache: false,
-         success: function(data) {
-           self.inner_components = self.inner_components.concat(data);
-         }
-       }).always(function(){
-           self.load_components()
-           setTimeout(self.load_icons, 2000)
-       })
+      this.fetchData();
    },
    methods: {
-     load_icons () {
-       var self = this
-       var components  = $('.k-toolbar .k-toolbar-item')
-       $.each(components, function(index, component){
-           self.inner_components[index].icon = component.getAttribute('icon')
-           self.inner_components[index].tooltip = component.getAttribute('tooltip')
-       })
-       self.$emit('update:components', self.inner_components)
+     async fetchData() {
+      try {
+        const response = await axios.get(this.url);
+        this.toolbarItems.push(...response.data)
+      } catch (err) {
+        console.error(err)
+      } finally {
```

Comment:
```suggestion
      } catch (err) {
        console.error(err)
      } finally {
```
@viniarck cant we just add any error detection with the catch block?

Reply:
ill also prototype it for the one of the napps and create an issue for all of the ones that will need changing.

Reply:
Ill tell you if it functions well once Im done

Reply:
> @viniarck cant we just add any error detection with the catch block?

Yes @HeriLFIU. A try catch works as you'd expect in a async context. In our code base we also have many sync funcs https://github.com/kytos-ng/mef_eline/blob/master/ui/k-info-panel/show_circuit.kytos#L381-L399, and currently as you know with jquery we've been using `request.done` and `request.fail` function callbacks. 

Maybe we should stick with `axios.then().catch().`? Since that will work for both async and sync funcs? https://axios-http.com/docs/handling_errors

Think about it and do some experiments, then we can stick with it. It's fine to mix both, but if one pattern suffices for the typical use cases we have, then it's easier to have that too since it makes easier to reason and understand, especially since production code is usually read many more times than it's written.